src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const logs = [];
+  let currentState = state;
+
```

Comment:
I suggest to create copy of state once at the begging and then mutate the copy
`const currentState = { ...state }`

Reply:
Do i need just create stateCopy before the loop and mutate it at the beginning ?